quicklens/src/main/scala/com/softwaremill/quicklens/package.scala

```diff
@@ -47,20 +48,43 @@ package object quicklens {
     }
   }
 
-  implicit class QuicklensEach[F[_], T](t: F[T])(implicit f: QuicklensFunctor[F, T, T]) {
+  implicit class QuicklensEach[F[_], T](t: F[T])(implicit f: QuicklensEachFunctor[F, T]) {
     @compileTimeOnly("each can only be used inside modify")
     def each: T = sys.error("")
   }
 
-  trait QuicklensFunctor[F[_], A, B] {
-    def map(fa: F[A])(f: A => B): F[B]
+  trait QuicklensEachFunctor[F[_], T] {
+    def each(fa: F[T])(f: T => T): F[T]
   }
-  implicit def optionQuicklensFunctor[A, B]: QuicklensFunctor[Option, A, B] =
-    new QuicklensFunctor[Option, A, B] {
-      override def map(fa: Option[A])(f: (A) => B) = fa.map(f)
+
+  implicit def optionQuicklensFunctor[T]: QuicklensEachFunctor[Option, T] =
+    new QuicklensEachFunctor[Option, T] {
+      override def each(fa: Option[T])(f: T => T) = fa.map(f)
+    }
+
+  implicit def traversableQuicklensFunctor[F[_], T](implicit cbf: CanBuildFrom[F[T], T, F[T]], ev: F[T] => TraversableLike[T, F[T]]) =
+    new QuicklensEachFunctor[F, T] {
+      override def each(fa: F[T])(f: T => T) = fa.map(f)
     }
-  implicit def traversableQuicklensFunctor[F[_], A, B](implicit cbf: CanBuildFrom[F[A], B, F[B]], e1: F[A] => TraversableLike[A, F[A]]): QuicklensFunctor[F, A, B] =
-    new QuicklensFunctor[F, A, B] {
-      override def map(fa: F[A])(f: (A) => B) = fa.map(f)
+
+  implicit class QuicklensAt[F[_], T](t: F[T])(implicit f: QuicklensAtFunctor[F, T]) {
+    @compileTimeOnly("at can only be used inside modify")
+    def at(idx: Int): T = sys.error("")
+  }
+
+  trait QuicklensAtFunctor[F[_], T] {
+    def at(fa: F[T], idx: Int)(f: T => T): F[T]
+  }
+
+  implicit def seqQuicklensFunctor[F[_], T](implicit cbf: CanBuildFrom[F[T], T, F[T]], ev: F[T] => SeqLike[T, F[T]]) =
+    new QuicklensAtFunctor[F, T] {
+      override def at(fa: F[T], idx: Int)(f: T => T) = {
+        val builder = cbf(fa)
+        for (i <- 0 to (fa.size - 1)) {
+          if (i == idx) builder += f(fa(idx))
```

Comment:
Because at each iteration you query the collection at the current index: `fa(idx)`. In case of list, that can take linear time